test/e2e/specs/editor/various/block-bindings.spec.js

```diff
@@ -229,133 +50,230 @@ test.describe( 'Block bindings', () => {
+									source: 'core/post-meta',
+									args: { key: 'text_custom_field' },
+								},
+							},
+						},
+					},
+				} );
 				const headingBlock = editor.canvas.getByRole( 'document', {
 					name: 'Block: Heading',
 				} );
 				await headingBlock.click();
 
 				// Alignment controls exist.
 				await expect(
-					page.getByRole( 'button', {
-						name: variables.labels.align,
-					} )
+					page
+						.getByRole( 'toolbar', { name: 'Block tools' } )
+						.getByRole( 'button', { name: 'Align text' } )
 				).toBeVisible();
 
 				// Format controls don't exist.
 				await expect(
-					page.getByRole( 'button', {
-						name: variables.labels.bold,
-					} )
+					page
+						.getByRole( 'toolbar', { name: 'Block tools' } )
+						.getByRole( 'button', {
+							name: 'Bold',
+						} )
 				).toBeHidden();
 
 				// Heading is not editable.
-				const isContentEditable =
-					await headingBlock.getAttribute( 'contenteditable' );
-				expect( isContentEditable ).toBe( 'false' );
+				await expect( headingBlock ).toHaveAttribute(
+					'contenteditable',
+					'false'
+				);
 			} );
 		} );
 
 		test.describe( 'Button', () => {
 			test( 'Should show the key of the custom field when text is bound', async ( {
 				editor,
 			} ) => {
-				await editor.insertBlock( variables.blocks.buttons.textOnly );
+				await editor.insertBlock( {
+					name: 'core/buttons',
+					innerBlocks: [
+						{
+							name: 'core/button',
+							attributes: {
+								text: 'button default text',
+								url: '#default-url',
+								metadata: {
+									bindings: {
+										text: {
+											source: 'core/post-meta',
+											args: { key: 'text_custom_field' },
+										},
+									},
+								},
+							},
+						},
+					],
+				} );
 				const buttonBlock = editor.canvas.getByRole( 'document', {
 					name: 'Block: Button',
 					exact: true,
 				} );
-				const buttonText = await buttonBlock.textContent();
-				expect( buttonText ).toBe( variables.customFields.textKey );
+				await expect( buttonBlock ).toHaveText( 'text_custom_field' );
 			} );
 
 			test( 'Should lock text controls when text is bound', async ( {
 				editor,
 				page,
 			} ) => {
-				await editor.insertBlock( variables.blocks.buttons.textOnly );
-				const buttonBlock = editor.canvas.getByRole( 'document', {
-					name: 'Block: Button',
-					exact: true,
+				await editor.insertBlock( {
+					name: 'core/buttons',
+					innerBlocks: [
+						{
+							name: 'core/button',
+							attributes: {
+								text: 'button default text',
+								url: '#default-url',
+								metadata: {
+									bindings: {
+										text: {
+											source: 'core/post-meta',
+											args: { key: 'text_custom_field' },
+										},
+									},
+								},
+							},
+						},
+					],
 				} );
+				const buttonBlock = editor.canvas
+					.getByRole( 'document', {
+						name: 'Block: Button',
+						exact: true,
+					} )
+					.locator( 'div' );
```

Comment:
Do you know why we need this `div` locator?

Reply:
In the editor, the button block are two nested divs:
```
<div aria-label="Block: Button">
  <div role="textbox" contenteditable="false"></div>
</div>
```

As I wanted to check if `contenteditable` is true/false, I needed to access the inner div.

Reply:
Could we instead access it by its role? `getByRole( 'textbox' )`

Reply:
I made this change as part of [this other pull request](https://github.com/WordPress/gutenberg/pull/58785).